backend/src/routes/api/groups-config/groupsConfigUtil.ts

```diff
@@ -0,0 +1,192 @@
+import { FastifyRequest } from 'fastify';
+import {
+  GroupConfigMapData,
+  GroupConfigMapDataProcessed,
+  GroupsConfig,
+  KubeFastifyInstance,
+} from '../../../types';
+import {
+  getGroupsConfigMapData,
+  getGroupsConfigMapName,
+  getAllGroups,
+  updateGroupsConfigMapData,
+} from '../../../utils/groupsUtils';
+import { getUserName } from '../../../utils/userUtils';
+import { isUserAdmin } from '../../../utils/adminUtils';
+import createError from 'http-errors';
+
+const SYSTEM_AUTHENTICATED = 'system:authenticated';
+
+export const getGroupsConfig = async (fastify: KubeFastifyInstance): Promise<GroupsConfig> => {
+  const { namespace } = fastify.kube;
+  const coreV1Api = fastify.kube.coreV1Api;
+  const customObjectsApi = fastify.kube.customObjectsApi;
+
+  try {
+    const groupsCluster = await getAllGroups(customObjectsApi);
+    const groupsConfigName = await getGroupsConfigMapName(coreV1Api, namespace);
+    const groupsData = (await getGroupsConfigMapData(
+      coreV1Api,
+      namespace,
+      groupsConfigName,
+    )) as GroupConfigMapData;
+    const groupsProcessed = processGroupData(groupsData);
+    const groupsConfigProcessed = processGroupConfig(groupsProcessed, groupsCluster);
+    if (
+      groupsData.admin_groups !== groupsConfigProcessed.groupsConfigMapData.admin_groups ||
+      groupsData.allowed_groups !== groupsConfigProcessed.groupsConfigMapData.allowed_groups
+    ) {
+      await updateGroupsConfigMapData(
+        coreV1Api,
+        namespace,
+        groupsConfigName,
+        groupsConfigProcessed.groupsConfigMapData,
+      );
+    }
+    return groupsConfigProcessed.groupsConfig;
+  } catch (e) {
+    fastify.log.error('Error retrieving group configuration: ' + e.toString());
+    const error = createError(500, 'Error retrieving group configuration');
+    throw error;
+  }
+};
+
+export const updateGroupsConfig = async (
+  fastify: KubeFastifyInstance,
+  request: FastifyRequest,
+): Promise<{ success: GroupsConfig | null; error: string | null }> => {
+  const customObjectsApi = fastify.kube.customObjectsApi;
+  const coreV1Api = fastify.kube.coreV1Api;
+  const { namespace, currentUser } = fastify.kube;
+
+  const userName = await getUserName(request, customObjectsApi, currentUser);
+  const isAdmin = await isUserAdmin(fastify, userName, namespace);
+
+  if (!isAdmin) {
+    const error = createError(403, 'Error updating groups, user needs to be admin');
+    throw error;
+  }
+
+  const groupConfigUpdated = request.body as GroupsConfig;
+  const adminConfig = groupConfigUpdated.adminGroups
+    .filter((group) => group.enabled)
+    .map((group) => group.name);
+  const userConfig = groupConfigUpdated.userGroups
+    .filter((group) => group.enabled)
+    .map((group) => group.name);
+
+  if (adminConfig.length === 0 || userConfig.length === 0) {
+    const error = createError(403, 'Error, groups cannot be empty');
+    throw error;
```

Comment:
ngl, it is a little weird we throw localized errors like this.

Reply:
mmm, yeah, I used to have error codes for these kind of situations, just for the frontend to render and localize them as pleased, not sure how to do it in our project right now.